compiler/damlc/tests/daml-test-files/ContractKeyNotVisible.daml

```diff
@@ -80,7 +83,46 @@ blindLookup = scenario do
   blind <- getParty "b" -- Blind
   _ <- submit sig do create Keyed with ..
   -- Blind party can't do positive lookup with maintainer authority.
+  cid <- submit sig $ create (Delegation sig blind)
   submit blind do
-    mcid <- createAndExercise (Delegation sig blind) LookupKeyed
+    mcid <- exercise cid LookupKeyed
     assert (isNone mcid)
     pure ()
+
+template LocalKeyVisibility
+  with
+    sig : Party
+    obs : Party
+  where
+    signatory sig
+    observer obs
+    choice LocalLookup : ()
+      controller obs
+      do create (Keyed sig)
+         Some _ <- lookupByKey @Keyed sig
+         pure ()
+
+    choice LocalFetch : ()
+      controller obs
+      do create (Keyed sig)
+         _ <- fetchByKey @Keyed sig
+         pure ()
+
+localLookup = scenario do
+  p1 <- getParty "p1"
+  p2 <- getParty "p2"
+  cid <- submit p1 $ create (LocalKeyVisibility p1 p2)
+  submit p2 $ exercise cid LocalLookup
+
+localFetch = scenario do
+  p1 <- getParty "p1"
+  p2 <- getParty "p2"
+  cid <- submit p1 $ create (LocalKeyVisibility p1 p2)
+  submit p2 $ exercise cid LocalLookup
```

Comment:
Here, you should exercise `LocalFetch`, right?

Reply:
good catch 57ee4f03436d8a323ce42868c6fdc3d9c079a00b